src/coreclr/jit/lowerxarch.cpp

```diff
@@ -4575,6 +4601,14 @@ void Lowering::ContainCheckStoreIndir(GenTreeStoreInd* node)
         MakeSrcContained(node, src);
     }
 
+    // If the source is a BSWAP, contain it on supported hardware to generate a MOVBE.
+    // Ensure that we are not double containing when the load has been contained.
+    if (comp->opts.OptimizationEnabled() && src->OperIs(GT_BSWAP, GT_BSWAP16) && !src->gtGetOp1()->isContained() &&
+        comp->compOpportunisticallyDependsOn(InstructionSet_MOVBE) && IsSafeToContainMem(node, src))
+    {
+        MakeSrcContained(node, src);
+    }
```

Comment:
The generated code would be slightly more efficient for BSWAP16 if you prefer to do `movbe` on the store instead of read in this case. The reason is that if we are doing a small write we can avoid normalizing after the bswap16.

I.e. you could do `src->gtGetOp1()->ClearContained()` instead. Up to you.